[READY] Making corpses rot and create miasma

Shdorsh opened the pull request:
[Changelogs]: # Soooo, I added a gas named miasma, which comes from rotting corpses, hoping to give atmos and virology a little bit more depth. It has a tiny chance to create a pretty non-hostile disease. If you want to bargain, I can set the chance even lower so it doesn't occur too much.

:cl: Shdorsh
add: A gas named miasma, that has a ridiculously low chance of infecting people with a very basic and benign disease
add: A gas reaction to remove miasma
tweak: Corpses emit a gas named miasma
tweak: Air alarms detect and automatically are set to scrub miasma
/:cl:

[why]: # Giving a little bit more depth to both atmos and viro, especially giving viro an incentive to work on vaccines for diseases and corpses to be either cloned earlier or removed/ stored properly in morgue/coffins/body bags or just plain burned.

Here's the gif: 
![miasmanew](https://user-images.githubusercontent.com/32726535/44438604-2e117500-a5c0-11e8-9496-0ae522fbad84.gif)

Shdorsh committed:
Adding miasma gastype

Shdorsh committed:
make miasma roll small chance of infecting and corpses rot

Shdorsh committed:
Merge pull request #7 from Shdorsh/patch-25

make miasma roll small chance of infecting and corpses rot

Shdorsh committed:
Add dry air sterilization to clear miasma

Shdorsh committed:
Merge pull request #8 from Shdorsh/patch-25

Add dry air sterilization to clear miasma

Shdorsh committed:
make air alarm scrub out miasma automatically

Shdorsh committed:
Merge pull request #9 from Shdorsh/patch-25

make air alarm scrub out miasma automatically

Militaires commented:
WE DF NOW BOYS

Shdorsh committed:
Storing corpse in morgue/coffin doesn't create miasma

fluffe9911 commented:
Everyday we get a bit closer to df being in ss13...again

Shdorsh committed:
Added more ways to block miasma and a small notice

Shdorsh commented:
Huh, what's df?

Shdorsh committed:
fixing some errors in the rot proc, since I accidentally copied the old bugged version

Shdorsh committed:
Code improvement, now waiting for Travis

Mark9013100 commented:
@Shdorsh Dorf Fortress

Shdorsh committed:
Added more effects to high miasma concentration

Bombpants commented:
Can we make dorfs a playable race?

Shdorsh committed:
Added delay to miasma production

As a result, miasma production goes up and partial pressure was unachievable back then, so I lowered it for all the effects.

Ratvar commented:
Does it mean reviving changelings will all cough and sneeze?

Bawhoppen commented:
Miasma should be a wizard power that causes diseases

Shdorsh commented:
I believe wizards already have quite the diversity in spells, no need to add some more. I once tried to code a totally modular spell that would upon choosing ask what kind of damage (oxy/brain/brute/stam/clone/burn/heal) it should apply, on whom (projectile/touch/adjacent tile/self) and how it should work (long reload/time needed to start up etc.) and giving the damage a multiplier for every weaker option taken, but it became too much spaghetti code. Might one day do that again for a future PR.

Ratvar commented:
Formaldehyde should prevent miasma formation or something.

Shdorsh commented:
Will be done, gimme a sec

Shdorsh committed:
Formaldehyde stops decay

Also still trying to manage that miasma side effect stuff

Shdorsh commented:
@Ratvar 
Makeshift supermatter noblium: deffo no. For diversity in gas types I made this one have low heat capacity. This stuff cools and heats pretty badly and has its temperature easily changed by other gases You might try stabilizing it with heat exchangers, but since miasma doesn't figure in the energy production equation, it has no influence afaik. I have yet to test out stuff like seeing whether sm can run purely on pressure. I know that it runs pretty well with co2 and n2o / n2 mixes without emitters. Also, at around 170°C, it becomes oxygen in a slightly exothermic reaction, which isn't too well supported by the sm. You can still try and post the results.

About the pressurization: basically, corpses bloat up, so having it produce a small amount of miasma isn't that bad.

No idea about the legion corpses lagging, didn't know that was an issue.

XDTM commented on code/modules/mob/living/carbon/life.dm:
```diff
@@ -217,6 +218,35 @@
+
+		if(prob(1 * miasma_partialpressure))
+			var/datum/disease/advance/miasma_disease = new/datum/disease/advance
+			miasma_disease.symptoms = miasma_disease.GenerateSymptoms(1,3)
```
`var/datum/disease/advance/miasma_disease = new/datum/disease/advance/random(max_symptoms, 3)` [either keep max_symptoms empty for a random amount or set it yourself]

Shdorsh replied:
The first 2 vars are needed for min level and max level. Some debug and coderbus symptoms I saw somewhere had level 0 and I don't want to include them.

XDTM replied:
Dont worry, random diseases exist for this exact purpose and can't pick level 0 symptoms

81Denton commented on code/modules/mob/living/carbon/life.dm:
```diff
@@ -217,6 +218,35 @@
+			if(0.1 to 0.5)
+				// At lower pp, give out a warning
+				if(prob(5))
+					to_chat(src, "<span class='notice'>There is an unpleasant smell in the air.</span>")
```
I hate myself for mentioning this, but these should probably have moodlet effects

Shdorsh replied:
I'll have a look at it later

Shdorsh replied:
Did it, but I'm unsure whether I did it right. Can you please have a look at it once the checks have all been done?

XDTM commented on code/modules/mob/living/carbon/life.dm:
```diff
@@ -217,6 +218,35 @@
+		if(prob(1 * miasma_partialpressure))
+			var/datum/disease/advance/miasma_disease = new/datum/disease/advance
+			miasma_disease.symptoms = miasma_disease.GenerateSymptoms(1,3)
+			miasma_disease.try_infect(src)
```
`ForceContractDisease(miasma_disease, TRUE, TRUE)`

try_infect ignores immunities

XDTM commented on code/modules/mob/living/carbon/life.dm:
```diff
@@ -245,6 +275,33 @@
 				return FALSE //to differentiate between no internals and active, but empty internals
 
+// Make corpses rot, emitting miasma
+/mob/living/carbon/proc/rot()
```
Maybe also check if the bodytemperature is very low, and that the mob_biotype is MOB_ORGANIC or MOB_UNDEAD since i doubt that golems would rot

Shdorsh replied:
Done

XDTM commented on code/modules/mob/living/carbon/life.dm:
```diff
@@ -245,6 +275,33 @@
+		return
+
+	// No decay if formaldehyde in corpse
+	if(typesof(/datum/reagent/toxin/formaldehyde) in reagents.reagent_list)
```
I'd suggest requiring a minimum amount of formaldehyde so you can't just sprinkle it with a dropper (since it never decays)

Shdorsh replied:
Done, require 20u at least.

Shdorsh committed:
fulfilling some reviews

Shdorsh committed:
Well done Travis

It's 9 in the morning, but I wanna see this through first

Shdorsh committed:
beginning of miasma canister

Shdorsh committed:
Added miasma canister

Shdorsh committed:
Merge pull request #12 from Shdorsh/patch-26

Adding miasma canister

Shdorsh committed:
Added miasma canister art

Shdorsh committed:
add odor moodlets

Shdorsh committed:
Merge pull request #13 from Shdorsh/patch-27

add odor moodlets

Shdorsh committed:
adding moodlets to miasma

Shdorsh committed:
simplifying reagent check

Shdorsh committed:
fixing indentation

Shdorsh committed:
fixing an indentation problem

Shdorsh committed:
Adding miasma tile effect

Shdorsh committed:
adding miasma tile effect

Shdorsh committed:
Deleting file in false placement

Shdorsh commented:
Still needs a little fix regarding how it handles miasma partial pressure, the moodlets don't seem to trigger, gimme a few minutes, folks

Grazyn commented:
So, goodbye hiding bodies?

Shdorsh commented:
@Grazyn Add formaldehyde or just send it in a nook outside the station or whatever. Should be do-able.

Tlaltecuhtli commented:
can u make corpses check for room temperature so u can actual make the chef coldroom a cold room for storing meat/dead ppl

Shdorsh committed:
changing miasma partial pressure for carbons

Shdorsh committed:
added effects to lungs

Shdorsh commented:
@Tlaltecuhtli It already checks for temperature, needing cryo to not generate miasma. I can however raise it up by a bit to let's say -15°C.

Shdorsh committed:
fixed a few errors

XDTM commented:
@Grazyn i fail to see the issue, discovering bodies due to the smell is a common trope and if someone's strategy for hiding bodies is to put them in a locker they deserve to get found

XDTM commented on code/modules/surgery/organs/lungs.dm:
```diff
@@ -270,12 +270,47 @@
+				if(prob(25))
+					to_chat(owner, "<span class='warning'>The stench of rotting carcasses is unbearable!</span>")
+					SEND_SIGNAL(owner, COMSIG_ADD_MOOD_EVENT, "nauseating_stench", /datum/mood_event/disgust/nauseating_stench)
+					owner.vomit()
```
this might be a permastun

Shdorsh committed:
Update lungs.dm

Grazyn commented:
@XDTM I'm asking because I suggested rotting and miasma years ago and I was told that it would make finding bodies too easy

XDTM commented:
Since this PR seems popular i guess we'll find out if that's the case

Shdorsh committed:
Making miasma effect slightly prettier

I am not a pixel artist, but at least I try

ExcessiveUseOfCobblestone commented:
Sometimes you lose to a man with a big nose

Shdorsh committed:
Raising miasma inhibiting temperature to -10°C

Yes, you can now store bodies in the cool room.

Shdorsh committed:
making sure air alarm is set for miasma

since it is a common waste product, air alarms should start with scrubbers having miasma set to scrub

Shdorsh committed:
setting scrubbers to scrub miasma on start

finally found this little setting

Shdorsh commented:
Well, I guess I'm done, private server tested and all.

Mickyan commented on code/modules/mob/living/carbon/life.dm:
```diff
@@ -245,6 +278,37 @@
+		return
+
+	// No decay if formaldehyde in corpse
+	if(reagents.has_reagent("formaldehyde", 20))
```
I suggest 15u just so it takes one full syringe per body and 2 bodies per 30u bottle

Shdorsh committed:
lowered formaldehyde amount needed

Shdorsh committed:
reaction fix

Shdorsh committed:
rebalanced the reaction

Now happens quicker and doesn't produce that much heat.

Qustinnus commented on code/datums/mood_events/needs_events.dm:
```diff
@@ -32,6 +32,16 @@
+/datum/mood_event/disgust/bad_smell
+	description = "<span class='warning'>You smell something horribly decayed inside this room.</span>\n"
+	mood_change = -3
+	timeout = 1800
```
If you want this to only affect people while you are inside of miasma; you should remove the timeout and use SEND_SIGNAL(src, COMSIG_CLEAR_MOOD_EVENT, "nauseating_stench") to remove it if there's not enough miasma.

this is fine too ofcourse but the description makes less sense if you keep the moodlet for 3 minutes after. :)

Qustinnus replied:
ditto for the other event ofc.

Tlaltecuhtli commented:
is the gas visible?

Qustinnus commented:
yes it is

epochayur commented:
should smelling dead-people gas lower your mood?

XDTM commented:
it does

Shdorsh commented:
will change moodlet as soon as I get back home

Shdorsh committed:
remove timeout to stench moods

Shdorsh committed:
Update life.dm

Shdorsh committed:
making mood stop with low/no miasma

AND waiting for travis

Shdorsh committed:
fixing variable naming error

Shdorsh committed:
fixing send signal for lungs

Shdorsh committed:
fixing send signal for carbon

Qustinnus commented on code/modules/mob/living/carbon/life.dm:
```diff
@@ -209,15 +210,62 @@
+				//At somewhat higher pp, warning becomes more obvious
+				if(prob(15))
+					to_chat(src, "<span class='warning'>You smell something horribly decayed inside this room.</span>")
+					SEND_SIGNAL(src, COMSIG_ADD_MOOD_EVENT, "bad smell", /datum/mood_event/disgust/bad_smell)
```
"bad smell" should be used for all the add_mood_events

Shdorsh replied:
Some smells are worse than others, which makes the mood go lower. It is also seen in how the side effect becomes stronger.

Qustinnus replied:
Yes, but you should only have one moodlet of a category at a time, the category would be "bad smell".

Right now, if the smell gets worse, youd get two moodlets.

Qustinnus commented on code/modules/mob/living/carbon/life.dm:
```diff
@@ -209,15 +210,62 @@
+	//Clear all moods if no miasma at all
+	else
+		END_SIGNAL(src, COMSIG_CLEAR_MOOD_EVENT, "bad smell")
+		END_SIGNAL(src, COMSIG_CLEAR_MOOD_EVENT, "nauseating_stench")
```
remove this and make the one above SEND_SIGNAL

Qustinnus commented on code/modules/mob/living/carbon/life.dm:
```diff
@@ -209,15 +210,62 @@
+				//Higher chance to vomit. Let the horror start
+				if(prob(25))
+					to_chat(src, "<span class='warning'>The stench of rotting carcasses is unbearable!</span>")
+					SEND_SIGNAL(src, COMSIG_ADD_MOOD_EVENT, "nauseating_stench", /datum/mood_event/disgust/nauseating_stench)
```
"bad smell"

Qustinnus commented on code/modules/mob/living/carbon/life.dm:
```diff
@@ -209,15 +210,62 @@
+				//Small chance to vomit. By now, people have internals on anyway
+				if(prob(5))
+					to_chat(src, "<span class='warning'>The stench of rotting carcasses is unbearable!</span>")
+					SEND_SIGNAL(src, COMSIG_ADD_MOOD_EVENT, "nauseating_stench", /datum/mood_event/disgust/nauseating_stench)
```
"bad smell"

Qustinnus commented on code/modules/surgery/organs/lungs.dm:
```diff
@@ -270,12 +270,60 @@
+				miasma_disease.symptoms = miasma_disease.GenerateSymptoms(1,3)
+				miasma_disease.try_infect(owner)
+
+			//Clearing out mood events if insufficient partial pressure
```
remove these 2 checks. they should be handled in the switch statement

Shdorsh replied:
you mind if I just keep the 2 moodlets and let another switch handle it so there's no conflicting 2 moodlets of the same type?

Qustinnus replied:
whats your discord handle ill explain how moodlets work

Qustinnus commented on code/modules/surgery/organs/lungs.dm:
```diff
@@ -270,12 +270,60 @@
+				miasma_disease.try_infect(owner)
+
+			//Clearing out mood events if insufficient partial pressure
+			if(miasma_pp < 25)
```
aka here and in the else case

Shdorsh committed:
fixing moodlet

this subsystem sure works in mysterious ways

Shdorsh committed:
fixing lung moodlets

smell u l8r

Shdorsh committed:
removing the 2 ifs

and looking if all moodlets are given to owner, not src

nicbn commented:
I didn't like the sprite
I think a green or purple colored water vapor sprite would be better

Shdorsh commented:
You're free to contribute a better sprite. For now, I'm busy trying to balance stuff

Shdorsh committed:
Trying to balance the sterilization reaction

No more accelerated superburn. This heats *slowly*, as it was supposed to.

Shdorsh commented:
It's done. Let the reviews/merge come in.

XDTM commented:
The random disease yhing i told you about

Shdorsh commented on code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm:
```diff
@@ -16,7 +16,7 @@
 	var/scrubbing = SCRUBBING //0 = siphoning, 1 = scrubbing
 
-	var/filter_types = list(/datum/gas/carbon_dioxide)
+	var/filter_types = list(/datum/gas/carbon_dioxide, /datum/gas/carbon_dioxide)
```
because I forgot to replace it with miasma I guess

Shdorsh committed:
Update vent_scrubber.dm

Shdorsh commented:
@mattroks101 done

Shdorsh commented:
We good now?

zamolxius commented:
can you make it visible

Shdorsh commented:
It is visible in somewhat high concentrations.

Kagus commented:
Do husks generate miasma, or does draining/charring a corpse help prevent it from getting all stinky-like?

Anyways, looks fun. We've needed some more random disease procs for a long while, although the randomized advance diseases are usually... Well... Generally somewhat short-lived, let's just say.

Is there any way to get the miasma disease proc to check for protective clothing such as bio suits, sterile masks and the like?

Shdorsh commented:
Charred corpses still emit some , but I'll tweak that soon. I believe the new proc one reviewer proposed for infection handles the infection resistance. I'll also post a link to a gif I took with th tile effect. Will take some timetho, got stuff planned today.

Shdorsh committed:
improved the disease generation

the random disease's new proc seems more efficient, so let's go with it

Shdorsh committed:
charred corpses don't produce miasma

there's nothing organic any saprophyte could possibly benefit from a carbonized carbon. also, replaced the disease proc, since there's better

Shdorsh commented:
There we go, charred corpses = no miasma. All done, I guess, except if any of you guys is good at spriting and can make a better looking canister, but that can always be PRed later.

Militaires commented:
why does there exist a canister for this wtf

Shdorsh commented:
For testing and gas labeling purposes. No sane mapper would put this on a map anyway.

Shdorsh commented:
You'd need a shitload of it tho, as it dissipates and just gets scrubbed out, plus it takes extremely long to get anything out of it as corpses generate it at a slow rate. And even then, it gives early warnings so that anyone would use internals anyway. In other words, I can't see this being weaponized effectively, except for adminbus.

Qustinnus commented:
@Shdorsh Now I think about it, why doesn't this use the pre-existing disgust system?

Shdorsh commented:
Would possibly raise and stack the disgust too much and we'd stack vomit to vomit.

Qustinnus commented:
then change the variables of how much you gain or change how much you need to start puking / how often you puke

Shdorsh commented:
Well, since I just tested with a reviewer of another code base and we figured it was pretty non-harmful, I guess I can add disgust level onto it.

Shdorsh committed:
adding disgust system

Shdorsh committed:
rebalancing

looking at the thresholds made me think that it was set too low, see negligible

ishitbyabullet commented:
So this won't put pressure in space ruins and only affects carbon mobs?

What if it's something like an away mission where there's no space or way to remove it. Won't there be this gas all over the ruin thus everyone would constantly get disease all because of one dead body?

ExcessiveUseOfCobblestone commented:
@ishitbyabullet you'd be wearing internals so nbd.

Shdorsh commented:
There already is a reaction that gets rid of it, folks, and it adds a _slight_ bit to the temperature. The last thing we would need is burning up the whole station because someone keeps hoarding all the corpses for their necrophilia fetish play.

Also, no gas in spess ruins since all you need is to be at -10°C to stop miasma generation.

And if it becomes too op, I'll just nerf its disgust generation anyway in the next PR. We're trying to build a **reasonable** game mechanic in here, folks.

XDTM commented:
I doubt it's ever going to be weaponized in its current form as long as NO2 exists, it won't be op. The part it shakes up is corpse disposal, and this is a great mechanic to make hiding corpses not as trivial.

Kagus commented:
And it'll make **perfectly legal** corpse cleanup and bodybags actually _mean_ something, which is also great.

ishitbyabullet commented:
I mean away missions.

Away missions can be on a planet, asteroid, disco funk party land, etc.

If they're on planets, though, then how do you prevent the entire planet being covered in miasama? I doubt people who make away missions want their map ruined by fucking bubble graphics giving everyone coughing fits.

PKPenguin321 commented on code/modules/atmospherics/machinery/portable/canister.dm:
```diff
@@ -137,6 +138,12 @@
 
+/obj/machinery/portable_atmospherics/canister/miasma
+	name = "miasma canister"
+	desc = "Miasma. Makes you wish your nose were blocked."
```
Try `Makes you wish your nose was blocked.` The `were` in the existing line sounds a bit awkward

PKPenguin321 replied:
You might be right

PKPenguin321 commented:
The gas sprite should be slowed down, very visually distracting at the speed in that gif

Shdorsh commented:
If the bubbling graphics are your biggest problem I could hardcap or softcap at a given concentration before it happens but that won't change much since it needs quite a bit to start bubbling anyway. The other thing that would be realistic is the availability of oxygen, which would probably stop miasma from being generated on lavaland etc.

Shdorsh committed:
Slowing down the bubbling

can we plz move on?

ishitbyabullet commented:
@Shdorsh what about non-lavaland planets.

Let's assume there is a map with no breachs to space and it is full of oxygen. It would be impossible to die from oxyloss unless you had done it to yourself through an empty oxy tank or something.

How would you prevent miasama from fucking over everything and making it look like you're under the sea full of water and shit?

XDTM commented:
If you come in a room full of rotten corpses it's gonna stink, ruin or not.

ishitbyabullet commented:
@XDTM You know what a room is, but does the game know?

If one dead body is in a 'room' that is 100 tiles wide and 100 tiles long then should it be 100% stinky?

Qustinnus commented:
@ishitbyabullet lol a non coder asking a coder if he knows about our code 10/10

Shdorsh commented:
If you're so afraid about that why don't you **read the fucking code**?

Do you even know how atmos works and stuff just gets thinned out in bigger rooms? Did you even rad how insignificantly much miasma is generated by a dead corpse? Did you even know that there needs to be a certain concentration before any real side effects happen due to miasma concentration?

For real, it's all in the code and in case you can't read code, don't bother acting like an authority on it.

Shdorsh commented:
Besides, you quasi-forced this comment out of me, no regrets.

Qustinnus commented:
@ishitbyabullet you're asking questions that were answered multiple times though or could be answered by looking at the code

XDTM commented:
No need to be aggressive, not everyone knows how to read dm code, and the amount of miasma released is not otherwise stated so miasma accumulation in corpse-filled ruins could have been a potential issue that @ishitbyabullet was pointing out.

XDTM commented:
By the way i suggest not making air alarms scrub miasma by default, as it will invalidate a lot of scenarios where corpses in hallways/rooms start miasma-ing the air.

ExcessiveUseOfCobblestone commented:
Rule 10, Not reading the code and looking like a dummy is a classic ss13 github experience.

Best part is it's not limited to people who do not KNOW how to read dm code.

Kagus commented:
Is it okay if I read some of the code and still look like a dummy?

ExcessiveUseOfCobblestone commented:
Yes and is encouraged to make this PR look important

Kagus commented:
Okay cool, thanks!

Iamgoofball commented:
no sprites?

ishitbyabullet commented:
If corpses generates miasma then I figured it would be possible that the dead bodies will eventually be able to fill anything that doesn't have an atmos system or scrubbers, especially if there's multiple corpses and it doesn't stop generating miasma. Doesn't matter if it's diluted since at some point infinite generation of one thing will have there be more of it than the thing diluting it (even if it takes forever to do so)

I was just concerned this would fuck over admin events/large away missions and cause people getting minor diseases at best or ocean bubbles ruining crap as it makes things look ugly at worst. That could discourage people if that was the case.

ExcessiveUseOfCobblestone commented:
Then those items will have to adapt or perhaps even incorporate the feature as an extra layer of difficulty. If you can think of any that might be problematic then feel free to test it yourself.

We shouldn't hold features hostage because people can't be arsed to maintain their ruins/missions/etc.

ishitbyabullet commented:
@ExcessiveUseOfCobblestone 
I just looked through the away missions in randomzlevels excluding VR. According to my quick lookthrough and brief memory on if any dead bodies are spawned, only one could have possible issues with marisma (research.dmm) however it seemed like it would actually make sense for it to be there. Not sure if wildwest.dmm had bodies but eh.

How would an away mission even adapt to this without adding in atmos/tiny air fans or changing a good chunk of it?

ExcessiveUseOfCobblestone commented:
internals or ways to remove the gas or just say "it's an away mission you should have prepped better enjoy ur diseases lol"

Shdorsh commented:
As I stated, there is already a way to remove miasma through heating it. If it's lavaland, it's possibly just going to get heated and turned into oxy. For other stuff, lungs remove miasma through breathing, and the script would make the chance be partial pressure * 0.1  % of getting a disease WHEN breathed in. Sitting around with internals doesn't give you any disease in it and if the partial pressure is like 0.001 due to how big the map is, it would mean you'd have 0.0001% chace to fall sick. Not a concern.

In order to even get the lowest alarm of noticing anything, it would require 1.5 pKa, which would be a bit of a challenge in the open or large rooms, unless people are right near the source and there are multiple corpses stacked there. To start really smelling something you'd require 10 pKa and to really have a low chance of throwing up, around 25 pKa, which would be a quarter of the normal atmospheric pressure. So really, I don't see any problem. If you really want to find out, testmerge it and then we talk. We're basically running around in circles here.

Shdorsh commented:
Besides, since you want gas mixture not to change in away missions, just make a PR that makes gas mixtures immutable on away mission turfs and we're done.

Kagus commented:
For being a constantly-burning magma planet covered in open-air lava rivers, Lavaland is remarkably... Temperate.

However, Lavaland atmospherics are in general _certifiably weird_ anyways, so there's not much risk of the miasma even mixing in properly to begin with.

Bawhoppen commented:
My bike shed away mission is filling up with miasma from corpses, please help.

ishitbyabullet commented:
@Shdorsh Actually, some away missions require gas mixtures so it wouldn't be as simple. I believe one of them is small yet has plasma so breaking windows is one of the worst things you can do. Another away mission has nitrogen outside the windows if I remember.

Thanks for answering my questions, though. Have a good day.

Shdorsh commented:
If you made a special subtype for those turfs inside the ruin to have immutable plasma gas, you'd solve the problem. Anyway, that's another PR.

ExcessiveUseOfCobblestone commented:
miascared?ma

adamkad1 commented:
re-add lava dwarves when?

Shdorsh committed:
restoring old scrubber functionality

Shdorsh committed:
mistakenly believed deleting this would only delete my changes in the code

here's the old functionality

Shdorsh committed:
Update and rename vent_scrubber to vent_scrubber.dm

Shdorsh commented on code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm:
```diff
@@ -1,331 +0,0 @@
-#define SIPHONING	0
-#define SCRUBBING	1
-
-/obj/machinery/atmospherics/components/unary/vent_scrubber
```
was my mistake for a sec, I just wanted to remove the miasma setting from the air alarm and thought the quickest way would be to remove all changes made to vent_scrubber. Accidentally undid the whole file. I'm somewhat new to github, but I try my best.

Shdorsh committed:
restituting the spacing

kevinz000 commented on code/modules/atmospherics/gasmixtures/gas_types.dm:
```diff
@@ -129,6 +129,13 @@ GLOBAL_LIST_INIT(nonreactive_gases, typecacheof(list(/datum/gas/oxygen, /datum/g
 
+/datum/gas/miasma
+	id = "miasma"
+	specific_heat = 0.01
```
*laughs in toxins*

Iamgoofball replied:
i like the idea of toxins using dead corpses for bomb fuel

granpawalton replied:
The low specific heat would make it a pretty good gas for fusion if you could harvest a decent amount but i feel like the work obtaining it would justify it.

davethwave replied:
Well if you can get a experimental cloner it would be kind of easy to accumulate the amount of bodies needed for such a thing, just create a closed area without air except a scrubber set to scrub miasma set up around the cloning thing so everytime someone pops out of the pod they suffocate to death and add to the miasma generation you have. That or just humanize a bunch of monkeys and kill them and do the same set up with a enclosed space.

cyclowns replied:
![image](https://user-images.githubusercontent.com/19853115/44884768-30d03200-ac72-11e8-9cdf-19b85152d459.png)
100% balanced

AnturK commented on code/modules/atmospherics/gasmixtures/reactions.dm:
```diff
@@ -433,4 +433,29 @@
+/datum/gas_reaction/miaster/react(datum/gas_mixture/air, datum/holder)
+	var/list/cached_gases = air.gases
+	// As the name says it, it needs to be dry
+	if(cached_gases[/datum/gas/water_vapor]/air.total_moles() > 0.1)
```
You're assuming gas contains this vapor without checking first.

Shdorsh replied:
Done.

Shdorsh committed:
adding water vapor check

optimumtact commented:
Looks okay, will keep it live for a day or two before merging.
--



Looks okay, will keep it live for a day or two before merging.
--



Looks okay, will keep it live for a day or two before merging.
--



Looks okay, will keep it live for a day or two before merging.
--



Looks okay, will keep it live for a day or two before merging.
--



Looks okay, will keep it live for a day or two before merging.
--



Looks okay, will keep it live for a day or two before merging.
--

vuonojenmustaturska commented on code/modules/atmospherics/gasmixtures/gas_types.dm:
```diff
@@ -129,6 +129,13 @@ GLOBAL_LIST_INIT(nonreactive_gases, typecacheof(list(/datum/gas/oxygen, /datum/g
 
+/datum/gas/miasma
+	id = "miasma"
+	specific_heat = 0.01
```
Change this to 20 to be in line with the standard air mix

Iamgoofball replied:
weird why can I resolve this

ExcessiveUseOfCobblestone replied:
I can too, we are all admins on coderbus now

Shdorsh committed:
changed miasma's specific heat to 20

no toxins memes

Shdorsh commented:
besides, any admemes done some roundend shenanigans with some miasma canisters? Just wanted to know so we have feedback on higher concentrations. It seems pretty much impossible to accumulate a lot of this, at least that's my atmos experience.

ExcessiveUseOfCobblestone commented:
if it's never realistically an issue what is the point of the gas? for the reference?

Shdorsh commented:
Well, it does add a small chance of becoming diseased, but it never occured to be so much that it ticked off the more drastic ill effects. I might need to lower the threshold.

Shdorsh commented:
Heck, I even hoarded 7 corpses or so and no message of foul smell popped up.

Shdorsh committed:
lowered threshold, due to near-unachievability of pressure

Shdorsh committed:
lowered threshold, due to near-unachievability of pressure

Shdorsh committed:
there's never going to be a canister onstation

unless it's debugging or adminbus, so you'll need this little bit more

Bawhoppen commented:
Don't say never. I bet someone will try to add a canister to a station they make. Then someone else will try and make the other stations "consistent" and add them all there too.

Shdorsh commented:
@Bawhoppen For now, I'll just have miasma canisters be filled, if anyone wants to test the more severe ill effects of it or so.

Bawhoppen commented:
It's not your fault anyways if they try and add it anyways. Was just joking around.

Kagus commented:
Clearly there needs to be a half-full miasma canister connected to the air scrubbers in bathrooms, to suck all the horrifying poo-gas out after the totally real bodily functions that spessmen need bathrooms for.

Shdorsh commented:
You can always use a normal emtpy tank anyway. Besides, toilets don't do anything, no miasma whatsoever.

Shdorsh commented:
Sooo... is all the testing done? Can we merge this now or do you want me to add another gas reaction and gas type?
